src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -190,6 +190,14 @@ public class LockFileBuilder
                     && (target.TargetFramework is FallbackFramework
                         || target.TargetFramework is AssetTargetFallbackFramework);
 
+                // This is used for a special-case warning we need to do when net6.0-maccatalyst selects a xamarin.ios asset.
+                MaccatalystFallback maccatalystFallback = null;
+
+                if (targetGraph.Framework.HasPlatform && targetGraph.Framework.Version.Major >= 6 && targetGraph.Framework.Platform.Equals("maccatalyst", StringComparison.OrdinalIgnoreCase))
```

Comment:
style: Given that this is relatively uncommon, extracting it in a separate method that just returns that object if necessary would improve the readability of what's already a very long method in CreateLockFile.